### PR TITLE
DAOS-11027 control: Set client secondary provider env

### PR DESCRIPTION
**Pull request by kjacque**
- Include a provider index in each provider's ClientNetHint in
  GetAttachInfo. This index corresponds to the position in the
  list of providers, starting with 0 for the primary provider.
- The provider index is used to set the environment variable
  CRT_SECONDARY_PROVIDER in the client lib.

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

File: src/mgmt/cli_mgmt.c [hunk 1]
```diff
@@ -407,13 +409,19 @@ int dc_mgmt_net_cfg(const char *name)
 	if (rc != 0)
 		D_GOTO(cleanup, rc = d_errno2der(errno));
 
+	sprintf(buf, "%d", info.provider_idx);
+	rc = setenv("CRT_SECONDARY_PROVIDER", buf, 1);
```
Comment (NiuYawei) [inline src/mgmt/cli_mgmt.c:413]: CRT_SECONDARY_PROVIDER is always set no matter what value the provider_idx is? And I'm not quite sure how to determine the 'provider_idx' when primary and secondary are using same provider?

Comment (frostedcmos) [reply src/mgmt/cli_mgmt.c:413]: I assume for primary provider the provider_idx will be 0? is that the case?
CART layer requires this to be set to non-0 for the clients running on the secondary cluster, and to 0/unset if running on the primary cluster

Comment (kjacque) [reply src/mgmt/cli_mgmt.c:413]: Yes, it will be 0 if using the primary provider.

Comment (kjacque) [reply src/mgmt/cli_mgmt.c:413]: And I'm not quite sure how to determine the 'provider_idx' when primary and secondary are using same provider?

Provider idx is tied to the order the parameters appear in the comma-separated provider string in the daos_server config file. So if you have a config: `provider: ofi+verbs,ofi+tcp`, idx 0 is ofi+verbs, idx 1 is ofi+tcp. If you wanted two TCP providers, you'd have: `provider: ofi+tcp,ofi+tcp`, so there would still be a provider at idx 1.

Comment (NiuYawei) [reply src/mgmt/cli_mgmt.c:413]: shouldn't we set CRT_SECONDARY_PROVIDER to 0 when the provider_idx is 0?

**Comment by kjacque** [comment]
The only failure was a fix Alex already landed to the branch, so I will land this one.

**Pull request merged**